### DISCUSSION TITLE
Multiple relations between models

### DISCUSSION THREAD

**Discussion by alucardu** [prisma/prisma#18039] [Q&A]
I have 3 models:

```
model User {
  id            Int         @id       @default(autoincrement())
  name          String
  session       Session?    @relation(fields: [sessionId], references: [id])
  sessionId     Int?
  @@unique([sessionId, name])
}

model Session {
  id            Int         @id       @default(autoincrement())
  name          String
  code          String      @unique
  players       User[]
}

model Message {
  id            Int         @id       @default(autoincrement())
  message       String
}
```

I want users to be able to create messages. 

A session can have many messages.
A user can have many messages.
A message can have one session.
A message can have one user.

I tried adding the relations:
```
model User {
  id            Int         @id       @default(autoincrement())
  name          String
  messages      Message[]   @relation(fields: [messageId], references: [id])
  messageId     Int
  session       Session?    @relation(fields: [sessionId], references: [id])
  sessionId     Int?
  @@unique([sessionId, name])
}

model Session {
  id            Int         @id       @default(autoincrement())
  name          String
  code          String      @unique
  players       User[]
  messages      Message[]
}

model Message {
  id            Int         @id       @default(autoincrement())
  message       String
  user          Session?    @relation(fields: [userId], references: [id])
  userId        Int
  session       Session?    @relation(fields: [sessionId], references: [id])
  sessionId     Int?
}
```
But this gives the (logical) error:

> Error validating model "Message": Ambiguous relation detected. The fields `user` and `session` in model `Message` both refer to `Session`. Please provide different relation names for them by adding `@relation(<name>).

I've read the docs > https://www.prisma.io/docs/concepts/components/prisma-schema/relations#disambiguating-relations but I cannot figure out the syntax/convention for the named fields.

**Comment by alucardu** [comment]
Made a small mistake in my `Message` model. `user` is defined as a `session`, should be `user`. Once I fixed that the error made sense and the fix was easy.

**Reply by nurul3101** [reply]
Thanks for letting us know, To confirm was this your final schema?
```prisma
model User {
  id        Int       @id @default(autoincrement())
  name      String
  messages  Message[]
  session   Session?  @relation(fields: [sessionId], references: [id])
  sessionId Int?

  @@unique([sessionId, name])
}

model Session {
  id       Int       @id @default(autoincrement())
  name     String
  code     String    @unique
  players  User[]
  messages Message[]
}

model Message {
  id        Int      @id @default(autoincrement())
  message   String
  user      User?    @relation(fields: [userId], references: [id])
  userId    Int
  session   Session? @relation(fields: [sessionId], references: [id])
  sessionId Int?
}

```

**Reply by alucardu** [reply]
No, the Prisma schema I'm working with looks like this:

```prisma
model User {
  id        Int       @id @default(autoincrement())
  name      String
  messages  Message[] @relation(name: "userMessage")
  messageId Int?
  session   Session?  @relation(fields: [sessionId], references: [id])
  sessionId Int?

  @@unique([sessionId, name])
}

model Session {
  id       Int       @id @default(autoincrement())
  name     String
  code     String    @unique
  players  User[]
  messages Message[]
}

model Message {
  id        Int      @id @default(autoincrement())
  message   String
  user      User?    @relation(name: "userMessage", fields: [userId], references: [id])
  userId    Int
  session   Session? @relation(fields: [sessionId], references: [id])
  sessionId Int?
}
```
I'm not sure that this actually works, I'm working on implementing the resolvers etc.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
